apps/gdalwarp_lib.cpp

```diff
@@ -2205,6 +2208,12 @@ static void AdjustOutputExtentForRPC( GDALDatasetH hSrcDS,
                     static_cast<int>(std::ceil(dfMaxX)) + nPadding - nWarpDstXOff);
                 nWarpDstYSize = std::min(nWarpDstYSize - nWarpDstYOff,
                     static_cast<int>(std::ceil(dfMaxY)) + nPadding - nWarpDstYOff);
+                if( nWarpDstXSize <= 0 || nWarpDstYSize <= 0 )
+                {
+                    CPLError(CE_Warning, CPLE_AppDefined,
+                             "No intersection between source extent defined by RPC and target extent");
+                    return false;
```

Comment:
How convinced are you that this should issue a warning?  We don't warn for other files provided to gdalwarp that do not overlap the target file do we?  Perhaps it should only be a debug message?

Reply:
changed to a debug message